.github/workflows/pyright.yml

```diff
@@ -0,0 +1,22 @@
+name: pyright
+
+on: [push]
+
+jobs:
+  pyright:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        version: ["3.10", "3.9", "3.8", "3.7"]
```

Comment:
you can skip 3.8 and 3.9, we suppose that if extremal version works they should also work